addon/components/paper-password.js

```diff
@@ -1,53 +1,62 @@
-import Component from '@ember/component';
-import { computed } from '@ember/object';
+import Component from '@glimmer/component';
 import { A } from '@ember/array';
 import { assert } from '@ember/debug';
 import zxcvbn from 'zxcvbn';
-import layout from '../templates/components/paper-password';
-
-export default Component.extend({
-  layout,
-  minStrength: 3,
-  strengthLabel: 'Password strength: ',
-  strengthLevels: A(['Very Poor', 'Poor', 'Fair', 'Good', 'Excellent']),
-  passwordErrorMessage: 'Please enter a stronger password.',
-  errors: A([]),
-  customValidations: A([]),
-
-  passwordStrength: computed('value', function() {
-    let password = this.value;
+
+export default class PaperPassword extends Component {
+  constructor() {
+    super(...arguments);
+
+    assert(
+      '<PaperPassword> requires an `onChange` action or null for no action.',
+      this.args.onChange !== undefined
+    );
+
+    // configure defaults
+    this.customValidations = this.args.customValidations || A([]);
+    this.errors = this.args.errors || A([]);
+    this.minStrength = this.args.minStrength || 3;
+    this.onChange = this.args.onChange || null;
+    this.passwordErrorMessage =
+      this.args.passwordErrorMessage || 'Please enter a stronger password.';
+    this.strengthLabel = this.args.strengthLabel || 'Password strength: ';
+    this.strengthLevels =
+      this.args.strengthLevels ||
+      A(['Very Poor', 'Poor', 'Fair', 'Good', 'Excellent']);
```

Comment:
I decided it was best to set the expected defaults for these on component creation.

It does mean that these specific arguments cannot be mutated after component creation but it feels like these would tend to be  hardcoded. For example, it would be odd to dynamically change the password strength level after component creation...

Can change later if we are so inclined, or the community can make an good argument for it. ✌️ 

Simplifies the parsing of defaults as otherwise they all need to be computed properties which ends up with a whole lot of getters wrappers:

```js
get passwordErrorMessage() {
  return this.args.passwordErrorMessage || 'Please enter a stronger password.';
}
```